src/components/DocumentControls/DocumentControls.js

```diff
@@ -0,0 +1,178 @@
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { saveAs } from 'file-saver';
+import { useTranslation } from 'react-i18next';
+
+import Button from 'components/Button';
+import './DocumentControls.scss';
+import getPageArrayFromString from 'helpers/getPageArrayFromString';
+import core from 'core';
+import selectors from 'selectors';
+import actions from 'actions';
+
+function getPageString(selectedPageArray, pageLabels) {
+  let pagesToPrint = '';
+  const sortedPages = selectedPageArray.sort((a, b) => a - b);
+  let prevIndex = null;
+
+  for (let i = 0; sortedPages.length > i; i++) {
+    if (sortedPages[i + 1] === sortedPages[i] + 1) {
+      prevIndex = prevIndex !== null ? prevIndex : sortedPages[i];
+    } else if (prevIndex !== null) {
+      pagesToPrint = `${pagesToPrint}${pageLabels[prevIndex]}-${pageLabels[sortedPages[i]]}, `;
+      prevIndex = null;
+    } else {
+      pagesToPrint = `${pagesToPrint}${pageLabels[sortedPages[i]]}, `;
+    }
+  }
+
+  return pagesToPrint.slice(0, -2);
+}
+
+const DocumentControls = props => {
+  const {
+    updateSelectedPage,
```

Comment:
Is it possible to just use the `setSelectedPageThumbnails` action instead of passing this prop down from `ThumbnailsPanel`